src/PostgreSql/custom/New-AzPostgreSqlFlexibleServer.ps1

```diff
@@ -0,0 +1,596 @@
+            }
+
+            if (!$PSBoundParameters.ContainsKey('Version')) {
+                $PSBoundParameters.Version = '12'
+            }
+
+            if (!$PSBoundParameters.ContainsKey('SkuTier')) {
+                $PSBoundParameters.SkuTier = 'GeneralPurpose'
+            }            
+
+            if ($PSBoundParameters.ContainsKey('AdministratorUserName')) {
+                $PSBoundParameters.AdministratorLogin = $PSBoundParameters['AdministratorUserName']
+                $null = $PSBoundParameters.Remove('AdministratorUserName')
+            }
+            else {
+                $PSBoundParameters.AdministratorLogin = Get-RandomName
+            }
+
+            $PSBoundParameters.CreateMode = [Microsoft.Azure.PowerShell.Cmdlets.PostgreSql.Support.CreateMode]::Default
+
+            # Handling Vnet & Subnet
+            $NetworkKeys = 'PublicAccess', 'Subnet', 'Vnet', 'SubnetPrefix', 'VnetPrefix'
+            $NetworkParameters = @{}
+            foreach($Key in $NetworkKeys){
+                if ($PSBoundParameters.ContainsKey($Key)){ 
+                    $NetworkParameters[$Key] = $PSBoundParameters[$Key]
+                    $null = $PSBoundParameters.Remove($Key)
+                }
+            }
+            $RequiredKeys = 'SubscriptionId', 'ResourceGroupName', 'Name', 'Location'
+            foreach($Key in $RequiredKeys){ $NetworkParameters[$Key] = $PSBoundParameters[$Key] }
+
+            if(!$NetworkParameters.ContainsKey('PublicAccess')){
+                $VnetSubnetParameters = CreateNetworkResource $NetworkParameters
+                $SubnetId = GetSubnetId $VnetSubnetParameters.ResourceGroupName $VnetSubnetParameters.VnetName $VnetSubnetParameters.SubnetName
+                $PSBoundParameters.DelegatedSubnetArgumentSubnetArmResourceId = $SubnetId
+                if ([string]::IsNullOrEmpty($PSBoundParameters.DelegatedSubnetArgumentSubnetArmResourceId)) {
+                    $null = $PSBoundParameters.Remove('DelegatedSubnetArgumentSubnetArmResourceId')
+                }
+            }
+            
+            $Msg = 'Creating PostgreSQL server {0} in group {1}...' -f $PSBoundParameters.Name, $PSBoundParameters.resourceGroupName
+            Write-Host $Msg
+            $Msg = 'Your server {0} is using sku {1} (Paid Tier). Please refer to https://aka.ms/postgresql-pricing for pricing details' -f $PSBoundParameters.Name, $PSBoundParameters.SkuName
+            Write-Host $Msg
+            $Server = Az.PostgreSql.internal\New-AzPostgreSqlFlexibleServer @PSBoundParameters
+
+            # Create Firewallrules
+            $FirewallRuleName = CreateFirewallRule $NetworkParameters
+
+            if (![string]::IsNullOrEmpty($FirewallRuleName)) {
+                $Server.FirewallRuleName = $FirewallRuleName
+            }
+            $Server.SecuredPassword =  $PSBoundParameters.AdministratorLoginPassword | ConvertTo-SecureString -AsPlainText -Force
```

Comment:
No Database name needed here?

Reply:
No database CRUD for PG now. It;s only recently implemented, that it is not included in the current swagger file